src-built-in/components/toolbarRefactored/toolbarHotkeys.ts

```diff
@@ -0,0 +1,109 @@
+const keys = FSBL.Clients.HotkeyClient.keyMap;
+
+/**
+ * Helper so I didn't have to rewrite the same 3 lines of code 6
+ * times.
+ * @param err Unlikely error; would be coming in when our hotkey is registered.
+ */
+const handleHotkeyRegistrationError = (err:any) => {
+	if (err) {
+		FSBL.Clients.Logger.error(`HotkeyClient.addGlobalHotkey failed, error:`, err);
+	}
+}
+/**
+	 * Function to bring toolbar to front (since dockable toolbar can be hidden)
+	 * The search input box will be open and any previous results will be displayed
+	 * @param {boolean} focus If true, will also focus the toolbar
+	 * @TODO Evaluate whether focus is necessary or if it's just vestigial.
+	 */
+const bringToolbarToFront = (focus = false) => {
+	finsembleWindow.bringToFront(null, (err:any) => {
+		if (err) {
+			FSBL.Clients.Logger.error(`bringToolbarToFront failed, error:`, err);
+		}
+
+		if (focus) {
+			finsembleWindow.focus();
+			// Leaving this line of code because clearly someone ran into something difficult.
```

Comment:
This comment needs to be more specific. It looks like a TODO item for you when the search component is complete. Are we redesigning the search bar? If so does the search bar need to be in focus when we bring up the toolbar using the hotkey?

Reply:
We aren't redesigning the search as far as I know. I have no idea what this line does but it seemed important. what exactly are you looking for in the comment?

Reply:
This line refocuses the search bar whenever someone uses the hotkey to bring up the toolbar. I'm saying if we want to keep this functionality we should refactor it when the search bar and the toolbar state management is setup.

Reply:
setActive false is refocusing the search bar? My comment was about the line of code below.

Reply:
@zona-ciq updated the comment. please take a look

Reply:
Resolved on the phone